Spot/Spot.swift

```diff
@@ -66,9 +68,15 @@ extension UIWindow {
         if let initialViewController = storyboard.instantiateInitialViewController() as? OrientationLockNavigationController {
             initialViewController.orientationToLock = UIDevice.current.orientation
             let window = UIApplication.shared.keyWindow
-            window?.rootViewController?.present(initialViewController, animated: true, completion: nil)
-            if let screenshotViewController = initialViewController.topViewController as? SpotViewController {
-                screenshotViewController.screenshot = screenshot
+            if var topController = window?.rootViewController {
+                while let presentedViewController = topController.presentedViewController {
+                    topController = presentedViewController
+                }
+                // topController should now be your topmost view controller
+                topController.present(initialViewController, animated: true, completion: nil)
```

Comment:
Rather than have the boolean check for whether we're on screen and track its state it might be nicer to just check whether the top view controller is one of Spots?